### PR TITLE
Emptying storage manure at the end of each storage time period

### LINKED ISSUE #1341: Storage manure not emptying at the end of each storage time period
Manure storage methods such as `slurry storage outdoor` and `slurry storage underfloor` are currently not updating the manure storage correctly at the end of the each `storage_time_period`. 

## Expected Behavior

Total volatile solids should drop back to zero at the end of each storage time period. 
VS SS 3x

Ideally the storage time period should overlap with the frequency when the manure is transported to the `soil and crop module` for field application (this is often how happens in reality). However, this implementation will take at least two steps. First, to reduce the amount of volatile solids in manure storage at the end of each storage time period (to zero) or when the `soil and crop module` calls for manure (depending on how much the `soil and crop module` needs). The latter is described in the issue #1272. This issue can hopefully address the first point.  

## Current Behavior

The amount of volatile solids does not change over simulation days. 
VS ADL

## Possible Solution

This feature works well in the `anaerobic lagoon` treatment, so what Loi and I did initially was to move the `_adjust_accumulated_output` function from the `anaerobic _lagoon.py` file to the `slurry_storage_outdoor.py`, `slurry_storage_underfloor.py` files. 

## Steps to Reproduce

1. Modify the `manure_management.json` input file for any of the two storage methods mentioned at the beginning. To see the effects of updating volatile solids with a short-term simulation, you can lower the `storage_time_period` to for example, 30 days. 
2. Update the animal input file with the corresponding manure management scenarios. 
3. Use the filter below to output accumulated volatile solids for each pen. 
```
^(?:AnaerobicDigestion_)?ManureTreatmentDailyOutput_Pen_.*storage_methane$
ManureHandlerDailyOutput_Pen_0_CALF.liquid_manure_total_volatile_solids
ReceptionPitDailyOutput_Pen_0_CALF.liquid_manure_total_volatile_solids
AnaerobicDigestion_ManureTreatmentDailyOutput_Pen_0_CALF.liquid_manure_total_volatile_solids
AnaerobicDigestion_ManureTreatmentDailyOutput_Pen_0_CALF.sludge_manure_total_volatile_solids
AnaerobicDigestion_ManureTreatmentDailyOutput_Pen_0_CALF.solid_manure_total_volatile_solids
ManureTreatmentDailyOutput_Pen_0_CALF.liquid_manure_total_volatile_solids
ManureTreatmentDailyOutput_Pen_0_CALF.sludge_manure_total_volatile_solids
ManureTreatmentDailyOutput_Pen_0_CALF.solid_manure_total_volatile_solids
Accumulated_ManureTreatmentDailyOutput_Pen_0_CALF.liquid_manure_total_volatile_solids
Accumulated_ManureTreatmentDailyOutput_Pen_0_CALF.sludge_manure_total_volatile_solids
Accumulated_ManureTreatmentDailyOutput_Pen_0_CALF.solid_manure_total_volatile_solids
ManureHandlerDailyOutput_Pen_1_GROWING.liquid_manure_total_volatile_solids
ReceptionPitDailyOutput_Pen_1_GROWING.liquid_manure_total_volatile_solids
AnaerobicDigestion_ManureTreatmentDailyOutput_Pen_1_GROWING.liquid_manure_total_volatile_solids
AnaerobicDigestion_ManureTreatmentDailyOutput_Pen_1_GROWING.sludge_manure_total_volatile_solids
AnaerobicDigestion_ManureTreatmentDailyOutput_Pen_1_GROWING.solid_manure_total_volatile_solids
ManureTreatmentDailyOutput_Pen_1_GROWING.liquid_manure_total_volatile_solids
ManureTreatmentDailyOutput_Pen_1_GROWING.sludge_manure_total_volatile_solids
ManureTreatmentDailyOutput_Pen_1_GROWING.solid_manure_total_volatile_solids
Accumulated_ManureTreatmentDailyOutput_Pen_1_GROWING.liquid_manure_total_volatile_solids
Accumulated_ManureTreatmentDailyOutput_Pen_1_GROWING.sludge_manure_total_volatile_solids
Accumulated_ManureTreatmentDailyOutput_Pen_1_GROWING.solid_manure_total_volatile_solids
ManureHandlerDailyOutput_Pen_2_CLOSE_UP.liquid_manure_total_volatile_solids
ReceptionPitDailyOutput_Pen_2_CLOSE_UP.liquid_manure_total_volatile_solids
AnaerobicDigestion_ManureTreatmentDailyOutput_Pen_2_CLOSE_UP.liquid_manure_total_volatile_solids
AnaerobicDigestion_ManureTreatmentDailyOutput_Pen_2_CLOSE_UP.sludge_manure_total_volatile_solids
AnaerobicDigestion_ManureTreatmentDailyOutput_Pen_2_CLOSE_UP.solid_manure_total_volatile_solids
ManureTreatmentDailyOutput_Pen_2_CLOSE_UP.liquid_manure_total_volatile_solids
ManureTreatmentDailyOutput_Pen_2_CLOSE_UP.sludge_manure_total_volatile_solids
ManureTreatmentDailyOutput_Pen_2_CLOSE_UP.solid_manure_total_volatile_solids
Accumulated_ManureTreatmentDailyOutput_Pen_2_CLOSE_UP.liquid_manure_total_volatile_solids
Accumulated_ManureTreatmentDailyOutput_Pen_2_CLOSE_UP.sludge_manure_total_volatile_solids
Accumulated_ManureTreatmentDailyOutput_Pen_2_CLOSE_UP.solid_manure_total_volatile_solids
ManureHandlerDailyOutput_Pen_3_LAC_COW.liquid_manure_total_volatile_solids
ReceptionPitDailyOutput_Pen_3_LAC_COW.liquid_manure_total_volatile_solids
AnaerobicDigestion_ManureTreatmentDailyOutput_Pen_3_LAC_COW.liquid_manure_total_volatile_solids
AnaerobicDigestion_ManureTreatmentDailyOutput_Pen_3_LAC_COW.sludge_manure_total_volatile_solids
AnaerobicDigestion_ManureTreatmentDailyOutput_Pen_3_LAC_COW.solid_manure_total_volatile_solids
ManureTreatmentDailyOutput_Pen_3_LAC_COW.liquid_manure_total_volatile_solids
ManureTreatmentDailyOutput_Pen_3_LAC_COW.sludge_manure_total_volatile_solids
ManureTreatmentDailyOutput_Pen_3_LAC_COW.solid_manure_total_volatile_solids
Accumulated_ManureTreatmentDailyOutput_Pen_3_LAC_COW.liquid_manure_total_volatile_solids
Accumulated_ManureTreatmentDailyOutput_Pen_3_LAC_COW.sludge_manure_total_volatile_solids
Accumulated_ManureTreatmentDailyOutput_Pen_3_LAC_COW.solid_manure_total_volatile_solids`
```
4. Use the code below to plot the accumulated volatile solids against simulation day. 
```
import pandas as pd
import matplotlib.pyplot as plt

# Read the CSV file into a pandas DataFrame
file_path = "/Users/haowenhu/Documents/GitHub/MASM/output/CSVs/default_saved_variables_csv_manure_storage_methane ts.txt_27-Feb-2024_Tue_10-55-55.csv"  # Replace "your_file.csv" with the path to your CSV file
data = pd.read_csv(file_path)

# Assuming 'column_name' is the name of the column you want to plot
column_name = 'Accumulated_ManureTreatmentDailyOutput_Pen_2_CLOSE_UP.liquid_manure_total_volatile_solids'
# column_name = 'ManureTreatmentDailyOutput_Pen_3_LAC_COW.storage_methane'

# Plot the column
plt.plot(data[column_name])
plt.xlabel('Simulation Day')
plt.ylabel('Accumulated liquid manure total volatile solids')
# plt.ylabel('Storage Methane')
plt.title('Plot of Volatile Solids from Close-up Pen')
# plt.title('Plot of Storage Methane from Lactating Pen')
plt.show()
```

### PR DESCRIPTION
**Pull request by allisterakun**
This PR is a **temporary fix** to the issue "Storage manure not emptying at the end of each storage time period" #1341.
`AnaerobicLagoon._adjust_accumulated_output()` is copied and pasted to be added to `SlurryStorageOutdoor` and `SlurryStorageUnderfloor` to empty the storage manure for each storage time period.
### Context

Issue(s) closed by this pull request: #1341 

- [x] The [changelog](https://docs.google.com/spreadsheets/d/1U-3igxdstHTFb6k5dVcE5-x9IL0r6q18eUiwwytKZyA/edit#gid=0) has been updated

### CODE REVIEW

**Comment by elle-andreen** [comment]
Both slurry outdoor and slurry underfloor appear to be emptying now! One thing I noticed though is that in addition to the accumulated quantities of nutrients resetting when the storage time runs out (as they should!), the accumulated manure treatment gas emissions (storage methane, ammonia, and nitrous oxide) also are resetting. I think the emissions should not be reset at the end of each storage time period; they're already emitted and up in the atmosphere, they (sadly) don't disappear along with the manure. So, a graph of the accumulated methane, ammonia, and nitrous oxide emissions should look basically like a continuously increasing line, not a zigzag line like the manure nutrients. 

We should eventually outline which variables in each manure treatment should and should not be reset when storage time is reached, although that's maybe outside the scope of this PR. For these two treatments specifically, I think the only relevant outputs that **do not** reset would be accumulated methane, ammonia, and nitrous oxide emissions. @KFosterReed can weigh in on 1. if emissions should be reset or not, and 2. if we should just worry about emissions for now and deal with variables in other treatments later.

**Comment by KFosterReed** [comment]
Ohh, great point Elle! Here is my thought on the Accumulated Emissions - if we want to make inference in the total emissions, we will most likely work with the sum of the daily outputs because we can specify the time period and almost always there is a spin up period that we want to ignore which we can't do automatically with the accumulated output anyway. 

TLDR - ignore the accumulated emissions for now and if you want to know what the accumulated emissions are, sum the daily output emissions

**Comment by allisterakun** [comment]
So... If I'm reading this correctly. The temporary fix is good for now, no further changes are needed for this PR.

@KFosterReed @elle-andreen Please confirm!

**Comment by KFosterReed** [comment]
I confirm if @elle-andreen doesn't find an error in my logic :)

**Review by KFosterReed — APPROVED** [review]

**Comment by elle-andreen** [comment]
T=

> Ohh, great point Elle! Here is my thought on the Accumulated Emissions - if we want to make inference in the total emissions, we will most likely work with the sum of the daily outputs because we can specify the time period and almost always there is a spin up period that we want to ignore which we can't do automatically with the accumulated output anyway.
> 
> TLDR - ignore the accumulated emissions for now and if you want to know what the accumulated emissions are, sum the daily output emissions

That makes sense to me!

**Review by elle-andreen — APPROVED** [review]
Thanks for your quick work on this Allister!
